### PR TITLE
Cleanup & Formatting

### PR DESCRIPTION
**Pull request by Xyaren**
reformatted in PEP 8 style
cleanup
added .editorconfig
fixed .dockerignore and .gitignore for venv
optimized imports
added actions for flake8 (linter for style & syntax) and pytest (unittests)

### CODE REVIEW

File: TS3Bot.py [hunk 1]
```diff
@@ -806,25 +815,20 @@ def servergroupaddperm(sgid, permsid, permvalue):
...
                         try:
-                            u = User(ts3conn, unique_id = a, ex_hand = signal_exception_handler)
+                            u = User(ts3conn, unique_id=a, ex_hand=signal_exception_handler)
                             tsdbid = u.ts_db_id
-                            _, ex = ts3conn.ts3exec(lambda tsc: tsc.exec_("setclientchannelgroup"
-                                                                    , cid = cinfo.get("cid")
-                                                                    , cldbid = tsdbid
-                                                                    , cgid = contactgroup.get("cgid"))
-                                            , signal_exception_handler)
+                            _, ex = ts3conn.ts3exec(lambda tsc: tsc.exec_("setclientchannelgroup", cid=cinfo.get("cid"), cldbid=tsdbid, cgid=contactgroup.get("cgid")),
```
Comment (ogrady) [inline TS3Bot.py:820]: This line appears to be 168 characters wide, which violates the new guideline of having `max_line_length = 120`, no?

File: Main.py [hunk 2]
```diff
@@ -108,37 +105,41 @@ def try_get(dictionary, key, lower = False, typer = lambda x: x, default = None)
...
-                        , ("Rising River", "Side", "Side")
-                        , ("Demons of Dawn", "DoD", "DoD")]
-
-            #for gname, gtag, ggroup in testguilds:
+            # BOT.setResetroster(ts3conn, "2020-04-01",
+            #                    red = ["the name with the looooong name"],
+            #                    green = ["another really well hung name", "len", "oof. tat really a long one duuuude"],
+            #                    blue = ["[DUST] dude", "[DUST] anotherone", "[DUST] thecrusty dusty mucky man"],
+            #                    ebg = [])
+            testguilds = [
```
Comment (ogrady) [inline Main.py:116]: Rather a todo, but this should obviously rather be in a separate test file, now that we are introducing actual unit tests.

File: Main.py [hunk 3]
```diff
@@ -28,27 +24,28 @@ def try_get(dictionary, key, lower = False, typer = lambda x: x, default = None)
...
             httpserver.start()
 
-            ipcIsPublic = os.getenv("IPC_PUBLIC","false").lower() in ['true', '1', 'y', 'yes']
+            ipcIsPublic = os.getenv("IPC_PUBLIC", "false").lower() in ['true', '1', 'y', 'yes']
             if ipcIsPublic:
                 log.warn("The IPC socket is open to the network, this is only ok in isolated and/or "
-                            "secure environments")
+                         "secure environments")
             """
             IPCS=ipc.TwistedServer(Config.ipc_port, ts3conn,
```
Comment (ogrady) [inline Main.py:50]: The `TwistedServer` should be gone by now. Is this PR based on an older commit?

File: TS3Bot.py (hunk 1 above)
Comment (Xyaren) [reply TS3Bot.py:820]: Limit for pythin files is 200.
120 seems to be a bit tight, and as everyone has widescreen dispalys nowadays, I choose 200.
Violations will be reported by the flake8 check

File: Main.py (hunk 3 above)
Comment (Xyaren) [reply Main.py:50]: just a leftover. Will clean that up.

**Pull request merged**